tests/prefetch_related/test_prefetch_related_objects.py

```diff
@@ -221,3 +223,73 @@ def test_prefetch_queryset(self):
 
         with self.assertNumQueries(0):
             self.assertCountEqual(book1.authors.all(), [self.author1, self.author2])
+
+    def test_prefetch_related_objects_with_various_iterables(self):
+        """
+        Test that prefetch_related_objects works with various iterables:
+        list, set, tuple, dict_values, empty iterable, and duplicate objects.
+        """
+        book = self.book1
+        # set
+        prefetch_related_objects({book}, "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+        # tuple
+        prefetch_related_objects((book,), "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+        # dict_values
+        d = {"a": book}
+        prefetch_related_objects(d.values(), "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+        # frozenset
+        prefetch_related_objects(frozenset([book]), "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+        # generator
+        prefetch_related_objects((x for x in [book]), "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+        # map
+        prefetch_related_objects(map(lambda x: x, [book]), "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+        # filter
+        prefetch_related_objects(filter(lambda x: True, [book]), "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+        # collections.deque
+        prefetch_related_objects(deque([book]), "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+
+        # custom iterable
+        class MyIterable:
+            def __iter__(self_inner):
+                yield book
+
+        prefetch_related_objects(MyIterable(), "authors")
+        self.assertCountEqual(
+            book.authors.all(), [self.author1, self.author2, self.author3]
+        )
+
+        # empty iterable
+        prefetch_related_objects([], "authors")
+        prefetch_related_objects(set(), "authors")
+        prefetch_related_objects((), "authors")
+        prefetch_related_objects(frozenset(), "authors")
+        prefetch_related_objects({}.values(), "authors")
+        prefetch_related_objects((x for x in []), "authors")
+        prefetch_related_objects(map(lambda x: x, []), "authors")
+        prefetch_related_objects(filter(lambda x: True, []), "authors")
+        prefetch_related_objects(deque([]), "authors")
```

Comment:
I would separate out the empty cases to a new test
```suggestion
    def test_prefetch_related_objects_with_empty_iterables(self):
        empty_cases = {
            "list": [],
            "set": set(),
            "tuple": (),
            "dict_values": {}.values(),
            "frozenset": frozenset(),
            "generator": (x for x in []),
            "map": map(lambda x: x, []),
            "filter": filter(lambda x: True, []),
            "deque": deque([]),
        }
        for case_type, case in empty_cases.items():
            with self.subTest(case=case_type):
                with self.assertNumQueries(0):
                    prefetch_related_objects(case, "authors")
```